File: Backend/TruyenCV/Models/Comment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TruyenCV.Models;

[Table("Comments", Schema = "dbo")]
public class Comment
{
    [Key]
    [Column("comment_id")]
    public int CommentId { get; set; }

    [Required, StringLength(450)]
    [Column("user_id")]
    public string ApplicationUserId { get; set; } = null!;

    [ForeignKey(nameof(ApplicationUserId))]
    public ApplicationUser ApplicationUser { get; set; } = null!;

    // Comment 1 trong 2: story OR chapter
    [Column("story_id")]
    public int? StoryId { get; set; }
    public Story? Story { get; set; }

    [Column("chapter_id")]
    public int? ChapterId { get; set; }
    public Chapter? Chapter { get; set; }

    [Required]
    [Column("content")]
    public string Content { get; set; } = null!;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

File: Frontend/truyencv_frontend/lib/models/comment.dart
class Comment {
  final int commentId;
  final String applicationUserId;
  final String? userName;
  final int? storyId;
  final int? chapterId;
  final String content;
  final DateTime createdAt;

  Comment({
    required this.commentId,
    required this.applicationUserId,
    this.userName,
    this.storyId,
    this.chapterId,
    required this.content,
    required this.createdAt,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      commentId: json['commentId'] as int,
      applicationUserId: json['applicationUserId'] as String,
      userName: json['userName'] as String?,
      storyId: json['storyId'] as int?,
      chapterId: json['chapterId'] as int?,
      content: json['content'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}

class CommentCreateDTO {
  final int? storyId;
  final int? chapterId;
  final String content;

  CommentCreateDTO({this.storyId, this.chapterId, required this.content});

  Map<String, dynamic> toJson() {
    return {
      if (storyId != null) 'storyId': storyId,
      if (chapterId != null) 'chapterId': chapterId,
      'content': content,
    };
  }
}

class CommentUpdateDTO {
  final String content;

  CommentUpdateDTO({required this.content});

  Map<String, dynamic> toJson() {
    return {'content': content};
  }
}
